examples/CSVImporter/CSVImporter_test.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"os"
+	"testing"
+	"github.com/TheJumpCloud/jcapi"
+)
+
+const (
+	testUrlBase string = "https://console.jumpcloud.com/api"
+	authUrlBase string = "https://auth.jumpcloud.com"
+)
+
+var testAPIKey string = os.Getenv("JUMPCLOUD_APIKEY")
+var testSystemID string = os.Getenv("JUMPCLOUD_SYSTEMID")
+
+
+func TestCSVImporter(t *testing.T) {
+	// Attach to JumpCloud
+	jc := jcapi.NewJCAPI(testAPIKey, testUrlBase)
+
+	// Fetch all users in JumpCloud
+	userList, err := jc.GetSystemUsers(false)
+
+	if err != nil {
+		t.Fatalf("Could not read system users, err='%s'\n", err)
+		return
+	}
+
+	// Fetch our system from JumpCloud
+	system, err := jc.GetSystemById(testSystemID, true)
+
+	if err != nil {
+		t.Fatalf("Could not read system info for ID='%s', err='%s'\n", testSystemID, err)
+		return
+	}
+
+	if system.Hostname == "" {
+		t.Fatalf("Could not read system info for ID='%s', err='%s'\n", testSystemID, err)
+		return
+	}
+
+	// Create a CSV record to add a test user
+	csvrec := []string{"Joe", "Smith", "js", "TheMan@jumpcloud.com", "", "", "T", "", ""}
+
+	// Process this request record
+	ProcessCSVRecord(jc, userList, csvrec)
```

Comment:
From this point forward in the test, I would consider refactoring into a separate function, and calling into it, so you can execute a bunch of different test cases with little additional effort.

This test encompasses the "happy" case, and doesn't exercise the error handling and recovery.